fastboot-protocol/src/protocol.rs

```diff
@@ -109,7 +116,7 @@ impl<'a> FastBootResponse {
             Err(FastBootResponseParseError::UnknownReply)
         } else {
             let resp = std::str::from_utf8(&bytes[0..4]).unwrap();
-            let data = std::str::from_utf8(&bytes[4..]).unwrap();
+            let data = std::str::from_utf8(bytes_slice_null(&bytes[4..])).unwrap();
```

Comment:
Please replace the unwrap with an error return instead (both this line and the previous one)